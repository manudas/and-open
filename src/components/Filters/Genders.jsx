import Button from '@material-ui/core/Button';
import ButtonGroup from '@material-ui/core/ButtonGroup';
import Typography from '@material-ui/core/Typography';

import {
    withStyles,
} from '@material-ui/core/styles';

import * as options from '../../data/userDataOptions';

const StyledButton = withStyles({
    root: {
        color: '#858585',
        borderColor: '#858585',
        '&:hover' : {
            color: '#4a4a4a',
            borderColor: '#4a4a4a',
            backgroundColor: 'rgba(133, 133, 133, 0.4)'
        },
        '&.selected': {
            backgroundColor: 'rgba(133, 133, 133, 0.6)'
        }
    },
})(Button);

const StyledTypography = withStyles({
    root: {
        marginBottom: '2em',
        color: '#9a9a9a',
        fontSize: '0.9rem',
        fontWeight: 'bold',
    },
})(Typography);

const Genders = ({
    selectedValue = null,
    setValue,
}) => {
    return (
        <div>
            <StyledTypography id="non-linear-slider" gutterBottom>
                Gender
            </StyledTypography>
            <ButtonGroup
                color="primary"
                aria-label="gender selector"
                data-testid="genders"
            >
                {
                    options.genders.map((gender, genderIndex) => {
                        return (
                                <StyledButton
                                    className={selectedValue === genderIndex ? 'selected' : null}
                                    key={gender}
                                    onClick={() => setValue(genderIndex)}
                                >
                                    {gender}
                                </StyledButton>
                        );
                    })
                }
                <StyledButton
                    className={selectedValue === null ? 'selected' : null}
                    onClick={() => setValue(null)}
                >
                    All
                </StyledButton>
            </ButtonGroup>
        </div>
    );
}

const StyledGenders = withStyles({
    root: {
        width: 300,
        margin: 10,
    },
})(Genders);
export default StyledGenders;
