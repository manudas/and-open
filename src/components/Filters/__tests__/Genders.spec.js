import {
    render,
    fireEvent,
} from '@testing-library/react';
import { act } from "react-dom/test-utils";

import Genders from '../Genders';

describe('Test suite for Genders', () => {

    beforeEach(() => {
        jest.useFakeTimers();
    });

    it('should render and match snapshot', () => {
        let renderer;
        act(() => {
            renderer = render(<Genders />);
            jest.runAllTimers();
        });
        const { container } = renderer;
        expect(container).toMatchSnapshot();
    });

    it('should call setValue function if some button is clicked', () => {
        const setGender = jest.fn();
        let renderer;
        act(() => {
            renderer = render(<Genders setValue={setGender} />);
            jest.runAllTimers();
        });

        const {
            queryByTestId,
        } = renderer;

        const gendersButtons = queryByTestId('genders');

        // Simulate clicking on a male filter
        fireEvent.click(gendersButtons.firstChild); // male

        expect(setGender).toHaveBeenCalledTimes(1)
    });
});
