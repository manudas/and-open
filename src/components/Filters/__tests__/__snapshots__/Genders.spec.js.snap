// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for Genders should render and match snapshot 1`] = `
<div>
  <div>
    <p
      class="MuiTypography-root WithStyles(ForwardRef(Typography))-root-2 MuiTypography-body1 MuiTypography-gutterBottom"
      id="non-linear-slider"
    >
      Gender
    </p>
    <div
      aria-label="gender selector"
      class="MuiButtonGroup-root"
      data-testid="genders"
      role="group"
    >
      <button
        class="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-3 MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-outlinedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Male
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-3 MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-outlinedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Female
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-3 MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary selected MuiButton-outlinedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          All
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
